EIPS/eip-1559.md

```diff
@@ -9,130 +9,107 @@ category: Core
-* `EIP1559_GAS_LIMIT` is the space available for EIP1559 transactions (EIP1559 gas pool) and it is a function of `BLOCK_GAS_TARGET` and blocks since `INITIAL_FORK_BLKNUM`.  It becomes equal to `BLOCK_GAS_TARGET * 2` for blocks >= `FINAL_FORK_BLKNUM`
-* The gas limit for the legacy gas pool is `BLOCK_GAS_TARGET - EIP1559_GAS_TARGET`.  As `EIP1559_GAS_TARGET` increases towards `BLOCK_GAS_TARGET`, gas is moved from the legacy pool into the EIP1559 pool until all of the gas is in the EIP1559 pool
-* At `block.number == INITIAL_FORK_BLKNUM`, let `EIP1559_GAS_LIMIT = (GASLIMIT / 2)` so that the gas maximum is split evenly between the legacy and EIP1559 gas pools
-* As `block.number` increases towards `FINAL_FORK_BLKNUM`, at every block we shift `1/MIGRATION_DURATION_IN_BLOCKS` gas from the legacy pool into the EIP1559 gas pool (effectively doubling it as it moves, since the EIP1559 gas target is half of the limit)
-* At `block.number >= FINAL_FORK_BLKNUM` the entire `BLOCK_GAS_TARGET` is assigned to the EIP1559 gas pool and the legacy pool is 0
-
-For the gas price:
-
-* We add a new field to the block header, `BASEFEE`
-  * `BASEFEE` is maintained under consensus by the ethash engine
-* At `block.number == INITIAL_FORK_BLKNUM` we set `BASEFEE = INITIAL_BASEFEE`
-* `BASEFEE` is set as follows
+* `LEGACY_GAS_LIMIT`: `BLOCK_GAS_TARGET - EIP1559_GAS_TARGET`. The maximum amount of gas legacy transactions can use in a given block.
+* `EIP1559_GAS_LIMIT`: `EIP1559_GAS_TARGET * 2`. The maximum amount of gas EIP-1559 transactions can use in a given block.
+* `BASE_FEE`: 16th item in the block header. Represents the amount of attoeth burned for every unit of gas a transaction uses.
+* `BASE_FEE_MAX_CHANGE_DENOMINATOR`: `8`
+* `BASE_FEE_MAX_CHANGE`: `CURRENT_BLOCK.parent.BASE_FEE / BASE_FEE_MAX_CHANGE_DENOMINATOR`. The maximum amount the `BASE_FEE` can change between blocks (either up or down).
+* `INITIAL_BASE_FEE` : 0.000000001 ETH (1 nanoeth)
+* `BASE_FEE_CHANGE`: `CURRENT_BLOCK.parent.BASE_FEE`. The amount that the BASE_FEE will increase in `CURRENT_BLOCK` relative to `CURRENT_BLOCK.parent`.
+
+### Process
+* At `block.number == INITIAL_FORK_BLOCK_NUMBER` we set `BASE_FEE = INITIAL_BASE_FEE`
+* `BASE_FEE` is set as follows
   * Let `delta = block.gas_used - TARGET_GASUSED` (possibly negative).
-  * Set `BASEFEE = PARENT_BASEFEE + PARENT_BASEFEE * delta // TARGET_GASUSED // BASEFEE_MAX_CHANGE_DENOMINATOR`
-  * Clamp the resulting `BASEFEE` inside of the allowable bounds if needed, where a valid `BASEFEE` is one such that `abs(BASEFEE - PARENT_BASEFEE) <= max(1, PARENT_BASEFEE // BASEFEE_MAX_CHANGE_DENOMINATOR)`
+  * Set `BASE_FEE = PARENT_BASE_FEE + PARENT_BASE_FEE * delta // TARGET_GASUSED // BASE_FEE_MAX_CHANGE_DENOMINATOR`
```

Comment:
We should explicitly define `//` as euclidean division somewhere

Reply:
I have a follow-up PR where I change the entire specification section into fully functional Python, so I'm hesitant to bother trying to define `//` here at the moment.  If my follow-up PR doesn't get merged (e.g., people don't like the change) then I agree, we should define that (I actually had to look it up myself).

Reply:
Sounds good! Heads up, both the Besu and Geth imps are using euclidean.